### DISCUSSION TITLE
Plans to export Aria Components context provider types?

### DISCUSSION THREAD

**Discussion by ProTip** [adobe/react-spectrum#4609] [General]
Looks like a new pattern is being tried out in Aria Components alongside slot props; context props.

Currently only `Provider` and `useContextProps` are exported from the package with a few missing like `ContextValue` type.

How is this working out? Are there plans to export all the relevant types so we can integrate into this system with custom components?

**Comment by snowystinger** [comment]
I don't know if I'm not following, but the Context types are all exported in React Aria Components https://github.com/adobe/react-spectrum/blob/main/packages/react-aria-components/src/index.ts#L29

Is there a specific problem you've run into?

**Comment by ProTip** [comment]
`ContextValue` which is used when constructing contexts is not exported from util: https://github.com/adobe/react-spectrum/blob/main/packages/react-aria-components/src/index.ts#L38 .

Edit: It's not a huge deal and I can work around it. Was just curious if there were plans to fully export the system for "public" use with custom context-aware components or if it's being held back as an implementation detail right now explicitly.

**Reply by snowystinger** [reply]
Ah, apologies, I had not followed. Thank you for clearing that up. I'll bring it up with the team.
